dashboard/generate_stats.py

```diff
@@ -0,0 +1,184 @@
+"""Generate the json data to feed the dashboard view
+Usage:
+    python generate_stats.py <output file>"""
+import csv
+import io
+import json
+import hashlib
+import os
+import time
+
+from boto.s3.connection import S3Connection
+from boto.s3.key import Key
+import requests
+
+from airtable.airtable import get_all_records
+
+S3_BUCKET = "dxe-backup"
+S3_BACKUP_DIR = "dashboard"
+S3_ACCESS_KEY = os.environ["AIRTABLE_BACKUP_AWS_ACCESS_KEY_ID"]
+S3_SECRET_KEY = os.environ["AIRTABLE_BACKUP_AWS_SECRET_ACCESS_KEY"]
+
+
+def chunks(l, n):
+    """Yield successive n-sized chunks from l."""
+    for i in xrange(0, len(l), n):
+        yield l[i:i+n]
+
+
+def links_to_ids(links):
+    """Uses the Facebook Graph API to get the group/page's id from their link"""
+
+    # This uses's the batch request feature
+    # (https://developers.facebook.com/docs/graph-api/making-multiple-requests)
+    # to make many logical requests within one http request, saving the overhead
+    # cost
+
+    ids = []
+    links = links[:49]
+    for links_subset in chunks(links, 50):  # batch limit is 50
+        reqs = ['{{"method":"GET", "relative_url":"{}"}}'.format(link) for link in links]
+        batch_req = '[' + ",".join(reqs) + ']'
+
+        r = requests.post(
+            "https://graph.facebook.com/v2.5",
+            data={"batch": batch_req},
+            headers={"Authorization": "Bearer {}".format(os.environ["FACEBOOK_APP_ACCESS_TOKEN"])}
+        )
+        data = r.json()
+        for res in data:
+            community = json.loads(res['body'])['id']
+            if community[0] == 'h':
```

Comment:
are FB ids [0-9] numbers? maybe community.startswith('http') would be more clear.